sljit_src/sljitWXExecAllocator.c

```diff
@@ -27,61 +27,91 @@
 /*
    This file contains a simple W^X executable memory allocator for POSIX systems.
 
-   It allocates a separate block for each code block and may waste a lot of memory
-   because it uses multiple (rounded up from the size requested) page (minimun 4KB)
-   sized blocks.
+   MAP_ANON is required (that is considered a feature) so make sure to adjust
+   your definitions accordingly or the code will fail to build.
 
-   It changes the page permissions as needed RW <-> RX and therefore if you will be
-   updating the code after it has been generated, need to make sure to block any
-   concurrent execution or could result in a segfault, that could even manifest in
-   a different address than the one that was being modified.
+   If your system doesn't support mapping of anonymous pages (ex: IRIX) it is
+   likely it also doesn't need this allocator and should be using the standard
+   one instead.
+
+   It allocates a separate block for each code block and may waste a lot of
+   memory because it uses multiple (rounded up from the size requested) page
+   sized (minimum 4KB) blocks.
+
+   It changes the page permissions (RW <-> RX) as needed and therefore, if you
+   will be updating the code after it has been generated, need to make sure to
+   block any concurrent execution, or could result in a segfault, that could
+   even manifest itself at a different address than the one that was being
+   modified.
 
    Only use if you are unable to use the regular allocator because of security
    restrictions and adding exceptions to your application is not possible.
 */
 
+#include <sys/types.h>
 #include <sys/mman.h>
-#include <unistd.h>
+
+#ifdef __NetBSD__
+#if defined(PROT_MPROTECT)
+#define SLJIT_PROT_WX PROT_MPROTECT(PROT_EXEC)
```

Comment:
All supported NetBSD version have `PROT_MPROTECT`.

Reply:
but the sljit codebase still supports unsupported versions of NetBSD until #58 is merged